src/lsp.rs

```diff
@@ -374,10 +443,30 @@ fn get_uri_from_arguments(arguments: &[serde_json::Value]) -> Url {
     }
 }
 
-pub async fn run<TLocalLinter: LocalLinter + 'static>(local_linter: TLocalLinter) {
+fn get_supported_language_language(uri: &Url) -> SupportedLanguageLanguage {
+    // TODO: how should this actually work?
```

Comment:
We presumably want this to "match" what tree-sitter-grep/`ignore` would decide but I'm not sure if there's a "direct" API exposed for that (basically mapping path -> `SupportedLanguage`)?